bayesian_logistic_regression.py

```diff
@@ -0,0 +1,365 @@
+import math
+import os
+import torch
+import torch.distributions.constraints as constraints
+import pyro
+from pyro.optim import Adam, SGD
+from pyro.infer import SVI, Trace_ELBO, config_enumerate, TraceEnum_ELBO, Predictive
+import pyro.distributions as dist
+from pyro.infer.autoguide import AutoDelta
+from pyro import poutine    
+from pyro.poutine import trace, replay, block
+from functools import partial
+import numpy as np
+import scipy.stats
+from pyro.infer.autoguide import AutoDelta
+from collections import defaultdict
+import matplotlib
+from matplotlib import pyplot
+from pyro.infer import MCMC, NUTS
+import pandas as pd
+import pickle
+from pyro.infer.autoguide import AutoDiagonalNormal
+import inspect
+from bbbvi import relbo, Approximation
+
+PRINT_INTERMEDIATE_LATENT_VALUES = False
+PRINT_TRACES = False
+
+# this is for running the notebook in our testing framework
+smoke_test = ('CI' in os.environ)
```

Comment:
Are we using this `smoke_test`?

Reply:
The Pyro GitHub repo has a continuous integration setup where they automatically run tests whenever anything is commited. They suggest that one uses this smoke_test flag to reduce the number of iterations of large loops, s.t. the tests run faster. E.g.: `n_steps = 2 if smoke_test else 12000`